### PR TITLE
End-to-end tests refactor

### PR DESCRIPTION
**Pull request by oreilles**
This will remove all dependencies on Dockerode in the e2e tests.
They used to require building the Directus image and spinning a Directus container for each test database.
With those changes, the local Directus code is used instead, making it much faster.

TODO:
- [x] Documentation 

In order to run the tests, you now have to run those commands from the repository's `/tests` directory:
* `npm run build`
* `docker compose -f tests/docker-compose.yml down -v` (clean up in case you ran the tests before)
* `docker compose -f tests/docker-compose.yml up -d --wait`
* `npm run test:e2e`

### CODE REVIEW

**Comment by rijkvanzanten** [comment]
(gh actions pls work 🤞🏻)

File: tests/e2e/config.ts [hunk 1]
```diff
@@ -128,7 +128,7 @@ const config: Config = {
 			DB_PASSWORD: 'secret',
 			DB_PORT: '6100',
 			DB_DATABASE: 'directus',
-			PORT: '49152',
+			PORT: '59152',
```
Comment (Oreilles) [inline tests/e2e/config.ts:131]: @licitdev Why is it needed ?

Comment (licitdev) [reply tests/e2e/config.ts:131]: These ports may already be in use by a root process: `remoted`, causing the tests to error out.
https://www.bleepingcomputer.com/forums/t/744822/question-regarding-remoted-its-purpose-and-link-local-addresses/

File: tests/e2e/get-dbs-to-test.ts [hunk 2]
```diff
@@ -1,24 +1,12 @@
 /** @TODO once Oracle is officially supported, enable it here */
 export const allVendors = ['mssql', 'mysql', 'postgres', /* 'oracle', */ 'maria' /*, 'sqlite3'*/, 'postgres10'];
```
[the remark this answers is not in the document]
Comment (Oreilles) [reply tests/e2e/get-dbs-to-test.ts:2]: Let's try that 👍

File: .github/workflows/e2e-tests.yml [hunk 3]
```diff
@@ -3,10 +3,6 @@ name: End-to-End Tests
 on:
   workflow_dispatch:
   workflow_call:
-    inputs:
-      should_skip:
```
[the remark this answers is not in the document]
Comment (Oreilles) [reply .github/workflows/e2e-tests.yml:7]: Nope, I moved the check up in `ci.yaml` instead for clarity:
https://github.com/directus/directus/pull/10968/files#diff-b803fcb7f17ed9235f1e5cb1fcd2f5d3b2838429d4368ae4c57ce4436577f03fR60

**Pull request merged**